app/vmctl/vm_native.go

```diff
@@ -109,6 +112,12 @@ func (p *vmNativeProcessor) do(ctx context.Context, f native.Filter, srcURL, dst
 
 func (p *vmNativeProcessor) runSingle(ctx context.Context, f native.Filter, srcURL, dstURL string) error {
 
+	matchWithFilters, err := buildMatchWithFilter(p.filter.Match, f.Match)
+	if err != nil {
+		return fmt.Errorf("failed to build export filters: %w", err)
+	}
+	f.Match = matchWithFilters
```

Comment:
`f.Match` is overwritten here. This looks OK in this particular case, since `f` arg is passed by value to `runSingle()` function, so the `f.Match` value at the caller side won't change. But this may break in the future if `f` arg will be passed by pointer to `runSingle()` function. Probably, it would be better creating a new variable here and using it afterwards? Something like the following:

```
filterFull := native.Filter{
   // ...
   Match: matchWithFilters,
}
```

Reply:
Moved function to the correct place